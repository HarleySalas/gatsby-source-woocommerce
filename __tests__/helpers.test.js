const { normaliseFieldName, processNode, } = require('../helpers/');

describe('Helpers:', () => {

  it('should normalize fieldname', () => {
    expect(normaliseFieldName('products/cateGorIes')).toEqual('productsCategories')
    expect(normaliseFieldName('products/categories')).toEqual('productsCategories')
  });

  it('should creates processNode', () => {
    const __type = 'wcProducts';
    const nodeWithOutType = {
      id: 1345,
      categories: [{ id: 23 }],
      wordpress_id: 1345,
    };
    const node = {
      __type,
      ...nodeWithOutType
    };
    const contentDigest = 19;
    const createContentDigest = jest.fn(() => contentDigest);
    const processNodeResult = processNode(createContentDigest, node);
    
    expect(createContentDigest).toBeCalled();
    expect(processNodeResult).toEqual({
      ...nodeWithOutType,
      parent: null,
      children: [],
      internal: {
        type: __type,
        contentDigest,
      },
    });

    expect(processNodeResult).toMatchSnapshot({
      id: expect.any(Number),
      categories: expect.any(Array),
      wordpress_id: expect.any(Number),
      parent: expect.any(Object),
      children: expect.any(Array),
      internal: {
        type: expect.any(String),
        contentDigest: expect.any(Number),
      },
    });
    
  });
  
});
