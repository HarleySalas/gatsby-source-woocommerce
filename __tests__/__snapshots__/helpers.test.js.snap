// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers: should creates processNode 1`] = `
Object {
  "categories": Any<Array>,
  "children": Any<Array>,
  "id": Any<Number>,
  "internal": Object {
    "contentDigest": Any<Number>,
    "type": Any<String>,
  },
  "parent": Any<Object>,
  "wordpress_id": Any<Number>,
}
`;
